File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineRouteError.java
package com.mapbox.services.android.navigation.v5.navigation;

import com.google.gson.annotations.SerializedName;

class OfflineRouteError {

  private final String status;
  @SerializedName("status_code")
  private final int statusCode;
  private final String error;
  @SerializedName("error_code")
  private final int errorCode;

  OfflineRouteError(String status, int statusCode, String error, int errorCode) {
    this.status = status;
    this.statusCode = statusCode;
    this.error = error;
    this.errorCode = errorCode;
  }

  public String getStatus() {
    return status;
  }

  public int getStatusCode() {
    return statusCode;
  }

  public String getError() {
    return error;
  }

  public int getErrorCode() {
    return errorCode;
  }
}

File: libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/model/OfflineRouteError.kt
package com.mapbox.navigation.route.onboard.model

import com.google.gson.annotations.SerializedName

internal data class OfflineRouteError(
    val status: String,
    @SerializedName("status_code") val statusCode: Int,
    val error: String,
    @SerializedName("error_code") val errorCode: Int
)
